src/Sylius/Behat/Page/Shop/Checkout/AddressingPageInterface.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Page\Shop\Checkout;
+
+use Sylius\Behat\Page\SymfonyPageInterface;
+use Sylius\Component\Core\Model\AddressInterface;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+interface AddressingPageInterface extends SymfonyPageInterface
+{
+    /**
+     * @throws \RuntimeException
+     */
+    public function chooseDifferentBillingAddress();
+
+    /**
+     * @param string $element
+     * @param string $message
+     * 
+     * @return bool
+     */
+    public function checkValidationMessageFor($element, $message);
+
+    /**
+     * @param string|null $firstName
+     */
+    public function specifyShippingAddressFirstName($firstName = null);
```

Comment:
How about we have only 2 methods: specifyShipping|BillingAddress(AddressInterface $address);

We can create empty address in context or fill only some fields. Would simplify the page interface a lot. And makes sense with the step structure suggestion from @Zales0123.